### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by hankim813**
- add more sensible inline comments
- cleanup iterators used

@jgershen @tsholmes @ndhoule 🙏

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -121,99 +124,90 @@ Optimizely.prototype.page = function(page) {
...
-          activeSections[activeSection.id] = true; // For deduping purposes
-        }
+        var activeVariation = variationIds[j];
+        var activeSection = variationIdsToSectionsMap[activeVariation];
+        if (activeSection) activeSections[activeSection.id] = activeSection.name;
       }
 
-      props.sectionName = sectionNames.join(', ');
-      props.sectionId = sectionIds.join();
+      props.sectionId = keys(activeSections).join(); // Not adding space for backward compat/consistency reasons since all IDs we've never had spaces
```
Comment (tsholmes) [inline lib/index.js]: These could be in a different order depending on the browser. We should probably sort the ids and names so there is a consistent value for the same active sections.

File: lib/index.js [hunk 2]
```diff
@@ -229,25 +223,26 @@ Optimizely.prototype.sendClassicDataToSegment = function(experimentState) {
...
-  each(function(audience) {
-    audienceIds.push(audience.id);
-    audienceNames.push(audience.name);
-  }, campaignState.audiences);
+  // Refomarting this data structure into hash map so concenating variation ids and names is easier later
+  var audiencesMap = foldl(function(results, audience) {
+    results[audience.id] = audience.name;
+    return results;
+  }, {}, campaignState.audiences);
+  var audienceIds = keys(audiencesMap).join();
```
Comment (tsholmes) [inline lib/index.js]: Same thing here with sorting for consistency.

**Review by tsholmes — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 1 above)
Comment (hankim813) [reply lib/index.js]: yep good call.

File: lib/index.js [hunk 3]
```diff
@@ -121,99 +124,93 @@ Optimizely.prototype.page = function(page) {
...
-  var variationIds = [];
-
-  // Makes concatenating easier later
-  each(function(variation) {
-    variationNames.push(variation.name);
-    variationIds.push(variation.id);
-  }, variations);
+  var context = { integration: optimizelyContext }; // backward compatibility
+
+  // Refomarting this data structure into hash map so concatenating variation ids and names is easier later
```
Comment (tsholmes) [inline lib/index.js]: s/Refomarting/Reformatting/

**Review by tsholmes — APPROVED** [review]
LGTM (with typo fixed)

**Comment by jgershen** [comment]
good stuff!

File: lib/index.js (hunk 3 above)
Comment (hankim813) [reply lib/index.js]: god damnit

**Pull request merged**